packages/react-router/modules/matchPath.js

```diff
@@ -1,5 +1,59 @@
 import pathToRegexp from "path-to-regexp";
 
+function isAbsolute(pathname) {
+  return pathname.charAt(0) === "/";
+}
+
+function ensureTrailingSlash(pathname) {
+  return hasTrailingSlash(pathname) ? pathname : pathname + "/";
+}
+
+function hasTrailingSlash(pathname) {
+  return pathname.charAt(pathname.length - 1) === "/";
+}
+
+function isMalformed(pathname) {
+  return pathname.slice(0, 3) === "...";
+}
+
+function refersToParentSegment(pathname) {
+  return pathname.slice(0, 3) === "../";
+}
+
+function refersToCurrentSegment(pathname) {
+  return pathname.slice(0, 2) === "./";
+}
+
+function resolvePath(pathname, base) {
+  if (pathname === undefined || isAbsolute(pathname)) {
+    return pathname;
+  }
+
+  if (isMalformed(pathname)) {
+    throw new Error("cannot resolve pathname: pathname is malformed");
```

Comment:
I'd include the pathname in the error message for easier debugging on the users end.